### PR TITLE
fix: update user data from IDP response

### PR DESCRIPTION
**Pull request by farmborough**
## :id:  
fixes gravitee-io/issues#7647

## :pencil2: A description of the changes proposed in the pull request
This fixes a bug where `UserAuthenticationService` was not parsing IDP responses for key fields such as email, username etc., and was only updating a user's additional properties. With this fix, a user profile will be updated whenever corresponding values are updated at the IDP.

### CODE REVIEW

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/auth/user/impl/UserAuthenticationManagerImpl.java [hunk 1]
```diff
@@ -256,7 +256,6 @@ private Maybe<UserAuthentication> loadUserByUsername0(Client client, Authenticat
                     Map<String, Object> additionalInformation = user.getAdditionalInformation() == null ? new HashMap<>() : new HashMap<>(user.getAdditionalInformation());
                     additionalInformation.put("source", authProvider);
                     additionalInformation.put(Parameters.CLIENT_ID, client.getId());
-                    ((DefaultUser) user).setAdditionalInformation(additionalInformation);
```
Comment (leleueri) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/auth/user/impl/UserAuthenticationManagerImpl.java:259]: removing this line may lead to regression as we test in line 256 if the additionalINformation is null. If so, we create a new HashMap but we do not set this map into the User object

**Review by leleueri — CHANGES_REQUESTED** [review]
In additon of the comment about the additionalInformation, I wondering if this PR is a duplicate of https://github.com/gravitee-io/issues/issues/7531

**Comment by farmborough** [comment]
> gravitee-io/issues#7531

I think it's a slightly different fix as this issue deals with updates originating from the IDP.

**Comment by leleueri** [comment]
After discussing with @farmborough same issue as https://github.com/gravitee-io/issues/issues/7531.
As 7531 will be merge in 3.15.x we close this one.

**Comment by farmborough** [comment]
After chatting with @leleueri this does indeed appear to be a duplicate and is already fixed by https://github.com/gravitee-io/issues/issues/7531.

**Pull request closed without merging**